src/uPlot.js

```diff
@@ -1681,8 +1681,17 @@ export default function uPlot(opts, data, then) {
 			_y = e.clientY - rect.top;
 		}
 		else {
-			_x = plotWidCss * (_x/_w);
-			_y = plotHgtCss * (_y/_h);
+			// check for value based syncs in the src plot
+			let [xKey, yKey] = src.cursor.sync.scales;
+			if (xKey != null)
+				_x = self.valToPos(src.posToVal(_x, xKey), xKey);
```

Comment:
we can definitely bypass `self.valToPos()` since we're not gonna need canvas pixels, let's call the lower-level `getXPos()` and `getYPos()` directly. `src.posToVal()` will probably stay (it's a bit cheaper).